### PR TITLE
Added trace log and error log.

### PR DESCRIPTION
**Pull request by makara**

### CODE REVIEW

File: server/middleware/errorHandlers/log.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+const logger = require('express-bunyan-logger');
+
+const lib = require('../../lib');
+
+module.exports = function(options) {
+  return logger.errorLogger({
```
Comment (makara) [inline server/middleware/errorHandlers/log.js]: Yeah I forgot it, but maybe later.

File: server/middleware/errorHandlers/log.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    logger: lib.logger(options),
+    immediate: false,
+    genReqId: lib.getTransactionId, // @see middleware/traceLog.js
+    levelFn
+  });
+};
+
+function levelFn(status, err) {
```
Comment (makara) [inline server/middleware/errorHandlers/log.js]: Yeah... looking.

File: server/middleware/traceLog.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+   */
+  lib.getTransactionId = function(req) {
+    return req[attr];
+  };
+
+  /**
+   * Log request.
+   */
+  const logRequest = logger({
+    logger: lib.logger(options),
```
Comment (makara) [inline server/middleware/traceLog.js]: No but I can see it's confusing. Refactoring.

File: server/middleware/errorHandlers/log.js (hunk 2 above)
Comment (makara) [inline server/middleware/errorHandlers/log.js]: Well no, they are different. Documenting.

**Comment by makara** [comment]
Merging for now, maybe some more later.

File: server/middleware.json [hunk 4]
```diff
@@ -37,6 +40,9 @@
     "loopback#urlNotFound": {}
   },
   "final:after": {
-    "strong-error-handler": {}
+    "./middleware/errorHandlers/log": {
+      "stream": "${logStream}"
+    },
```
Comment (haishanh) [inline server/middleware.json]: I think this should be

```json
"./middleware/errorHandlers/log": {
  "params": {
    "stream": "${logStream}"
  }
},
```

Comment (makara) [inline server/middleware.json]: Nice catch. Fixing.

**Pull request merged**